balances/src/impl_fungibles.rs

```diff
@@ -0,0 +1,180 @@
+use super::*;
+use encointer_primitives::common::PalletString;
+use frame_support::{
+	inherent::Vec,
+	traits::{
+		fungibles::InspectMetadata,
+		tokens::{DepositConsequence, WithdrawConsequence},
+	},
+};
+use sp_runtime::traits::Zero;
+
+impl<T: Config> fungibles::InspectMetadata<T::AccountId> for Pallet<T> {
+	fn name(_asset: &Self::AssetId) -> Vec<u8> {
+		PalletString::from("Encointer").into()
+	}
+
+	fn symbol(_asset: &Self::AssetId) -> Vec<u8> {
+		PalletString::from("ETR").into()
+	}
+
+	fn decimals(_asset: &Self::AssetId) -> u8 {
+		// Our BalanceType is I64F64, so the smallest possible number is 2^-64 = 5.42101086242752217003726400434970855712890625 × 10^-20
+		//  and an upper bound is 2^63 + 1 = 9.223372036854775809 × 10^18
+		// so we chose 18 decimals and lose some precision but can prevent overflows that way.
+		18u8
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn balance_type_to_fungible_balance(
+		asset: <Pallet<T> as fungibles::Inspect<T::AccountId>>::AssetId,
+		balance: BalanceType,
+	) -> <Pallet<T> as fungibles::Inspect<T::AccountId>>::Balance {
+		assert!(balance >= BalanceType::from_num(0));
+		let decimals = Self::decimals(&asset);
+
+		let bits = balance.to_bits();
+		let mut result: u128 = 0;
+
+		result = result + (bits >> 64) as u128 * 10u128.pow(decimals as u32);
+
+		result = result +
+			(BalanceType::from_bits((bits as i64) as i128) *
+				BalanceType::from_num(10u128.pow(decimals as u32)))
+			.to_num::<u128>();
+		result
+	}
+
+	pub fn fungible_balance_to_balance_type(
+		asset: <Pallet<T> as fungibles::Inspect<T::AccountId>>::AssetId,
+		fungible_balance: <Pallet<T> as fungibles::Inspect<T::AccountId>>::Balance,
+	) -> BalanceType {
+		let decimals = Self::decimals(&asset);
+		let mut result: BalanceType = BalanceType::from_num(0);
+
+		result = result +
+			BalanceType::from_num(
+				((fungible_balance << 64) >> 64) as f64 / (10i128.pow(decimals as u32) as f64),
+			);
+
+		result = result +
+			BalanceType::from_num(fungible_balance >> 64) *
+				BalanceType::from_num(
+					2i128.pow(64) as f64 / 10i128.pow(decimals as u32) as f64,
+				);
+		result
+	}
+}
+
+impl<T: Config> fungibles::Inspect<T::AccountId> for Pallet<T> {
+	type AssetId = CommunityIdentifier;
+	type Balance = u128;
```

Comment:
I think this is a fundamental misunderstanding. I think what you did works, but is much too complicated. We should use `BalanceType = EncointerBalanceType` here, so we can spare all those conversions.

Reply:
Ok, I see. This is not possible, it needs to be a scalar type implementing `AtLeast32Bit`.

Reply:
the problem was that it needed to be unsigned